### ISSUE TITLE
Unable to override resolve_type in UnionTypes

### ISSUE THREAD

**Issue by nostophilia** [rmosolgo/graphql-ruby#471]
In my Rails application I'm dealing with polymorphic associations with a UnionType. When I tried to override `resolve_type` in the UnionType, "NoMethodError - undefined method `resolve_type=' for FoobarType:GraphQL::UnionType" was raised. But I saw [UnionType still accepts_definitions of :resolve_type](https://github.com/rmosolgo/graphql-ruby/blob/master/lib/graphql/union_type.rb#L27), while it seems that this has been deprecated actually #290. 

But I wonder why. The official `GraphQL.js` is still having `resolveType?` in `InterfaceType` and `UnionType` [according to their doc](http://graphql.org/graphql-js/type/#graphqluniontype) which makes more sense to me as we may want to apply different resolve_type logics per type. It will be more flexible if we can have both schema level and type level.

**Comment by rmosolgo** [comment]
Originally, I followed `graphql-js`'s design for `resolve_type`. But I realized that it was possible to have a serious bug:

It was possible to provide `resolve_type` functions which returned _different_ GraphQL types for the same Ruby object. 

When that happened, query execution was very unpredictable! For this reason, I moved `resolve_type` to a single schema-level function. However, as you noticed, I forgot to remove the `accepts_definition` on unions! I've just removed it in b449175. 

Is there a case where type-level `resolve_type` does something that a schema-level `resolve_type` cannot do?
